crates/subspace-runtime-primitives/src/lib.rs

```diff
@@ -57,6 +57,13 @@ pub fn maximum_normal_block_length() -> BlockLength {
     BlockLength::max_with_normal_ratio(MAX_BLOCK_LENGTH, NORMAL_DISPATCH_RATIO)
 }
 
+/// The maximum recursion depth we allow when parsing calls.
+/// This is a safety measure to avoid stack overflows.
+///
+/// Deeper nested calls can result in an error, or, if it is secure, the call is skipped.
+/// (Some code does unlimited heap-based recursion via `nested_utility_call_iter()`.)
+pub const MAX_CALL_RECURSION_DEPTH: u32 = 10;
```

Comment:
This is something I have brought it earlier in the issue discussion. We do not want a simple constant as this is error prone. When we mistakenly updates this value and clients get upgraded, all upgraded client would use new limit while unupgraded will use older value. This will leads to fraud proofs.

Please, as suggested in the disucssion, client should pull this value from runtime call. If the call is unavaible, should fallback to decoding witout limit until runtime provides such function to ensure consistency across all clients

Reply:
Probably unnecessary.

The client side is using `OpaqueExtrinsic` it should not have the need (or even have the ability) to decode to the concrete `RuntimeCall`. Because the `Block` used on the client side is `opaque::Block` and it is fixed in the binary while the `Block` and the `RuntimeCall` in the runtime are defined by the runtime code which is stored in the state and can be upgraded, thus dynamically change.